evap/staff/tests/test_importers.py

```diff
@@ -181,34 +182,54 @@ def setUpTestData(cls):
         baker.make(CourseType, name_de="Vorlesung", import_names=["Vorlesung", "V"])
         Degree.objects.filter(name_de="Bachelor").update(import_names=["Bachelor", "B. Sc."])
         Degree.objects.filter(name_de="Master").update(import_names=["Master", "M. Sc."])
+        cls.default_excel_content = excel_data.create_memory_excel_file(excel_data.test_enrollment_data_filedata)
+        cls.create_existing_course(cls)
+
+    def create_existing_course(self):
+        self.existing_course = baker.make(
+            Course,
+            name_de="Schütteln",
+            name_en="Shake",
+            semester=self.semester,
+            type=CourseType.objects.get(name_de="Vorlesung"),
+            degrees=[Degree.objects.get(name_de="Bachelor")],
+            responsibles=[
+                baker.make(
+                    UserProfile,
+                    email="123@institution.example.com",
+                    title="Prof. Dr.",
+                    first_name="Christoph",
+                    last_name="Prorsus",
+                )
+            ],
+        )
+        self.existing_course_evaluation = baker.make(Evaluation, course=self.existing_course)
 
     def test_valid_file_import(self):
-        excel_content = excel_data.create_memory_excel_file(excel_data.test_enrollment_data_filedata)
-
         success_messages, warnings, errors = EnrollmentImporter.process(
-            excel_content, self.semester, None, None, test_run=True
+            self.default_excel_content, self.semester, None, None, test_run=True
         )
-        self.assertIn("The import run will create 23 courses/evaluations and 23 users:", "".join(success_messages))
+        self.assertIn("The import run will create 23 courses/evaluations and 22 users:", "".join(success_messages))
         # check for one random user instead of for all 23
         self.assertIn("Ferdi Itaque (789@institution.example.com)", "".join(success_messages))
         self.assertEqual(errors, {})
-        self.assertEqual(warnings, {})
+        self.assertNotEqual(warnings, {})
```

Comment:
Wait, why isn't `warnings` empty anymore? These tests should definitely not change, right? (I mean, this test is called `valid_file_import`)

edit: or is the excel file just wrong, according to the new logic?

Reply:
Resolving, explanation in https://github.com/e-valuation/EvaP/pull/1683#discussion_r816165654.